tests/ci/ci_test.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+ARTIFACT_DIR="${ARTIFACT_DIR:-/tmp/artifacts}"
+INFO="INFO"
+log_info(){
+    echo "$(date "+%Y-%m-%d %H:%M:%S") [$INFO] $1"
+}
+check_status(){
+    res=$1
+    namespace=$2
+    log_info "Checking status of $res inside $namespace"
+    oc rollout status $res -n $namespace --timeout=5m || (must_gather "$ARTIFACT_DIR" && exit 1)
```

Comment:
Do we need a subshell here? 
```suggestion
    oc rollout status $res -n $namespace --timeout=5m ||  { 
        must_gather "$ARTIFACT_DIR"
        return 1
      }
```

Reply:
Not mandatory. We can use `{}` along with `||` as well as it will also allow executing multiple commands if the `oc rollout status` fails